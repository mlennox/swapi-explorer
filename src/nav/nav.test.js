import React from "react";
import Nav from "./index";
import { render, asFragment, fireEvent } from "react-testing-library";
import "jest-dom/extend-expect";

describe("nav : ", () => {
  test("should render", () => {
    const { asFragment } = render(<Nav />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("click link should do something", () => {
    const { getByText, asFragment } = render(<Nav />);
    const peopleLink = getByText("People");
    fireEvent.click(peopleLink);
    expect(asFragment()).toMatchSnapshot();
  });
});
