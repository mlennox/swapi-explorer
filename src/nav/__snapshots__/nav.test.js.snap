// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav :  click link should do something 1`] = `
<DocumentFragment>
  <nav>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Home
    </a>
    <a
      href="/people"
    >
      People
    </a>
    <a
      href="/planets"
    >
      Planets
    </a>
    <a
      href="/species"
    >
      Species
    </a>
    <a
      href="/ships"
    >
      Ships
    </a>
    <a
      href="/vehicles"
    >
      Vehicles
    </a>
    <a
      href="/films"
    >
      Films
    </a>
  </nav>
  <h1>
    Home
  </h1>
</DocumentFragment>
`;

exports[`nav :  should render 1`] = `
<DocumentFragment>
  <nav>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Home
    </a>
    <a
      href="/people"
    >
      People
    </a>
    <a
      href="/planets"
    >
      Planets
    </a>
    <a
      href="/species"
    >
      Species
    </a>
    <a
      href="/ships"
    >
      Ships
    </a>
    <a
      href="/vehicles"
    >
      Vehicles
    </a>
    <a
      href="/films"
    >
      Films
    </a>
  </nav>
  <h1>
    Home
  </h1>
</DocumentFragment>
`;
